VOLDEV-186: Whitelist Gadget-* to make the gadget extension usable

onei opened the pull request:
Also whitelists Gadgets-*, which applies to messages defined by the extension itself, thus allowing users to alter gadgets-definition as well.

@Grunny

onei committed:
wfMsg cleanup

onei committed:
add prefixes to verbatim ext and whitelisted messages

mszabo-wikia commented:
CR: https://wikia-inc.atlassian.net/browse/SUS-1548

mszabo-wikia commented on extensions/Gadgets/Gadgets_body.php:
```diff
@@ -87,7 +87,10 @@ public static function getPreferences( $user, &$preferences ) {
-				$section = wfMsgExt( "gadget-section-$section", 'parseinline' );
+				// begin wikia change
+				// wfMsg cleanup
+				$section = wfMessage( "gadget-section-$section" )->parse();
```
We should consider either adding separate use for each possible value of `$section`, or at least list the possible values in comments - helps when searching for usage of messages

onei replied:
In this case, the name of the message is user defined allowing them to group similar gadgets under appropriate heads on gadgets-definition and Special:Gadgets. If you wanted a list of them, you'd have to look them up on Special:PrefixIndex on the wiki the extension is installed on or something similar.

For reference, the messages defined by the extension are prefixed with gadgets-*.

mszabo-wikia commented on extensions/Gadgets/SpecialGadgets.php:
```diff
@@ -188,7 +208,10 @@ public function showExportForm( $gadget ) {
-			. Xml::submitButton( wfMsg( 'gadgets-export-download' ) )
+			// begin wikia change
+			// wfMsg cleanup
+			. Xml::submitButton( wfMessage( 'gadgets-export-download' )->escaped() )
```
This function escapes its parameter, so no need for double escaping here

Also this is inside a special page so we can use `$this->msg` to set proper request context

onei replied:
Done in 21954ee

onei committed:
Add ticket to change comments + CR fixes:
- Use ->msg instead of WfMessage on special pages
- Use plain instead of escaped for message output when escaping is handled by Xml::submitButton

macbre commented:
Ok, let's merge this one into a branch in our upstream repository. Once it passes QA we can merge it into our dev.
